packages/firestore/test/unit/util/async_queue.test.ts

```diff
@@ -209,6 +209,28 @@ describe('AsyncQueue', () => {
     await queue.drain();
     expect(completedSteps).to.deep.equal([1, 2]);
   });
+
+  it('Schedules operaions with respect to shut down', async () => {
+    const queue = new AsyncQueue();
+    const completedSteps: number[] = [];
+    const doStep = (n: number): Promise<number> =>
+      defer(() => {
+        const result = completedSteps.push(n);
+        return result;
```

Comment:
Since you're not using the result, you could drop this return and change this to return `Promise<void>`.